custom_md_loader/classes/MetadataApexApiLoader.cls

```diff
@@ -0,0 +1,230 @@
+	public static boolean success1 {get;set;}
+
+    public MetadataApexApiLoader() {
+    	this.mdDeployStatus = new MetadataApexApiLoader.MetadataDeployStatus();
+    	this.callback = new MetadataDeployCallback(); 
+    } 
+    
+    public MetadataApexApiLoader.MetadataDeployStatus getMdDeployStatus() {
+    	return this.mdDeployStatus;
+	}
+	
+	public MetadataApexApiLoader.MetadataDeployCallback getCallback() {
+    	return this.callback;
+	}
+    
+    public override void migrateAsIsWithObjCreation(String csName, String cmtName) {
+		List<MetadataResponse.Message> messages = new List<MetadataResponse.Message>();
+		messages.add(new MetadataResponse.Message(100, 'Not Supported!!!'));
+		response.setIsSuccess(false);
+	    response.setMessages(messages);  
+    }
+    
+    public override void migrateAsIsMapping(String csName, String cmtName) {
+    	super.migrateAsIsMapping(csName, cmtName);
+    	buildResponse();
+    }
+    
+    public override void migrateSimpleMapping(String csNameWithField, String cmtNameWithField) {
+    	super.migrateSimpleMapping(csNameWithField, cmtNameWithField);
+    	buildResponse();
+    }
+    
+    public override void migrateCustomMapping(String csName, String cmtName, String mapping) {
+    	super.migrateCustomMapping(csName, cmtName, mapping);
+    	buildResponse();
+    }
+    
+    private void buildResponse() {
+    	if(response.IsSuccess()) {
+			List<MetadataResponse.Message> messages = new List<MetadataResponse.Message>();
+			messages.add(new MetadataResponse.Message(100, 'Migration In Progress... Job Id: ' + getMdDeployStatus().getJobId()));
+			response.setIsSuccess(true);
+		    response.setMessages(messages);  
+		}
+    }
+    
+    public override void migrate(MetadataMappingInfo mappingInfo) {
+    	
+    	System.debug('MetadataApexApiLoader.migrate -->');
+    	try{
+    		Map<String, Schema.DescribeFieldResult> descFieldResultMap = mappingInfo.getSrcFieldResultMap();
+			String typeDevName = mappingInfo.getCustomMetadadataTypeName()
+						.subString(0, mappingInfo.getCustomMetadadataTypeName().indexOf(AppConstants.MDT_SUFFIX));
+	        List<Metadata.CustomMetadata> records = new List<Metadata.CustomMetadata>();            
+			for(sObject csRecord : mappingInfo.getRecordList()) {
+			
+	            Metadata.CustomMetadata customMetadataRecord = new Metadata.CustomMetadata();
+	            customMetadataRecord.values = new List<Metadata.CustomMetadataValue>();
+	            
+	            if(csRecord.get(AppConstants.CS_NAME_ATTRIBURE) != null) {
+	            	String strippedLabel = (String)csRecord.get(AppConstants.CS_NAME_ATTRIBURE);
+	            	String tempVal = strippedLabel.substring(0, 1);
+	                
+	                if(tempVal.isNumeric()) {
+	                	strippedLabel = 'X' + strippedLabel;
+	                }
+	            	strippedLabel = strippedLabel.replaceAll('\\W+', '_').replaceAll('__+', '_').replaceAll('\\A[^a-zA-Z]+', '').replaceAll('_$', '');

```

Comment:
Can you doc the replacement strategy in either UI or Readme?

Reply:
yeap